fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/RecordQuery.java

```diff
@@ -170,11 +173,43 @@ public Builder toBuilder() {
         return new Builder(this);
     }
 
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof RecordQuery)) {
+            return false;
+        }
+        final RecordQuery that = (RecordQuery)o;
+        return isSortReverse() == that.isSortReverse() &&
+               removeDuplicates == that.removeDuplicates &&
+               getRecordTypes().equals(that.getRecordTypes()) &&
+               Objects.equals(getAllowedIndexes(), that.getAllowedIndexes()) &&
+               Objects.equals(queryabilityFilter, that.queryabilityFilter) &&
```

Comment:
I don't think there's any effort currently to make `IndexQueryabilityFilter` instances have a solid `equals`. Since it's a `@FunctionalInterface`, they can easily be some random lambda or even worse closure.

Reply:
As per discussion I removed the `@FunctionalInterface` and insist on a proper implementation of `queryHash()`

Reply:
So, since this is `@Nonnull`, we can go with `queryabilityFilter.equals(that.queryabilityFilter)`, right?

Reply:
Yes, we can! Fixed!